### PR TITLE
[Merged by Bors] - feat port:Data.Polynomial.RingDivision

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2763
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2763
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2763~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Polynomial/RingDivision.lean [hunk 1]
```diff
@@ -355,32 +364,36 @@ section CommRing
 
 variable [CommRing R]
 
+/- Porting note: due to a diamond, introduced
```
Comment (jcommelin) [inline Mathlib/Data/Polynomial/RingDivision.lean:368]: I don't understand how this porting note relates to neighbouring lines or the diff with mathport-output.

File: Mathlib/Data/Polynomial/RingDivision.lean [hunk 2]
```diff
@@ -126,49 +126,53 @@ section NoZeroDivisors
...
+instance : NoZeroDivisors R[X] where 
+  eq_zero_or_eq_zero_of_mul_eq_zero h := by
+    rw [← leadingCoeff_eq_zero, ← leadingCoeff_eq_zero]
     refine' eq_zero_or_eq_zero_of_mul_eq_zero _
-    rw [← leading_coeff_zero, ← leading_coeff_mul, h]
+    rw [← leadingCoeff_zero, ← leadingCoeff_mul, h]
+
+/- Porting note: I could not find this and Lean didn't do it automatically. Perhaps it belongs
+elsewhere. -/
+theorem WithBot.some_eq_nat_cast (n : ℕ) : WithBot.some n = (n : WithBot ℕ) := rfl
```
[the remark this answers is not in the document]
Comment (mattrobball) [reply Mathlib/Data/Polynomial/RingDivision.lean]: Thanks!

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**